lib/LedgerSMB/PGTimestamp.pm

```diff
@@ -0,0 +1,223 @@
+
+package LedgerSMB::PGTimestamp;
+
+=head1 NAME
+
+LedgerSMB::PGTimestamp - Timestamp handling and serialization to database
+
+=head1 DESCRIPTION
+
+This class handles formatting and mapping between the DateTime module and
+PostgreSQL. It provides a handler for the timestamp datatype.
+
+The type behaves internally as a Datetime module.
+
+=cut
+
+use strict;
+use warnings;
+use base qw(PGObject::Type::DateTime);
+
+use Carp;
+use DateTime::Format::Strptime qw(strptime);
+
+use LedgerSMB::App_State;
+use LedgerSMB::Magic
+    qw( MONTHS_PER_QUARTER YEARS_PER_CENTURY FUTURE_YEARS_LIMIT );
+
+use overload (
+    fallback => 1,
+    q{""}    => '_stringify',
+    );
+
+__PACKAGE__->register(registry => 'default',
+                      types => ['timestamp', 'timestamptz']);
+
+sub _stringify {
+    my $self = shift;
+
+    # Stringify without the 'T' in the middle
+    #   which is the same way Pg stringifies
+    return $self->strftime('%F %T');
+}
+
+=head1 SUPPORTED FORMATS
+
+### TO DOCUMENT
+
+=cut
+
+
+=head1 CONSTRUCTOR SYNTAX
+
+LedgerSMB::PgDate->new({ date => DateTime->new(year => 2012, day => 31, month =>
```

Comment:
Should this be LedgerSMB::PGTimestamp rather than LedgerSMB::PGDate?

Reply:
As the constructor is private anyway: simply removed the invocation docs -- you're not supposed to invocate it anyway.